src/main/java/lt/verbus/repository/BankAccountRepository.java

```diff
@@ -0,0 +1,119 @@
+package lt.verbus.repository;
+
+import lt.verbus.exception.EntityNotFoundException;
+import lt.verbus.model.*;
+
+import java.sql.*;
+import java.util.List;
+
+public class BankAccountRepository extends GenericRepository<BankAccount> {
+
+    public BankAccountRepository(Connection connection) throws SQLException {
+        super(connection, "bank_account");
+    }
+
+    public List<BankAccount> findAll() throws SQLException {
+        return super.findAll();
+    }
+
+    public BankAccount findByIban(String iban) throws SQLException, EntityNotFoundException {
+        return super.findByUniqueCode("iban", iban);
+    }
+
+    public List<BankAccount> findAllBelongingTo(Object object) throws SQLException {
+        String query = ("SELECT * FROM bank_account WHERE ");
+        if (object instanceof User) {
```

Comment:
Jei kode naudojamas instanceof tai toki koda galima pakoreguoti ir padaryti be. Siulau sukurti du skirtingus metodus panaikinant generic Object parametra nurodant tikslaiai klase, o pati Select'a iskelti i konstanta.